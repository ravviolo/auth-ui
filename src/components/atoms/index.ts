export { Button } from './Button/Button';
export { Checkbox } from './Checkbox/Checkbox';
export { Divider } from './Divider/Divider';
export { Heading } from './Heading/Heading';
export { IconButton } from './IconButton/IconButton';
export { InputField } from './InputField/InputField';
export { InputLabel } from './InputLabel/InputLabel';
export { TextButton } from './TextButton/TextButton';
