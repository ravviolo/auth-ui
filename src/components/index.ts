export * from './templates/LoginTemplate/LoginTemplate';
export * from './templates/SignUpTemplate/SignUpTemplate';
