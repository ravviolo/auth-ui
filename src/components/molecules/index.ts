export { AuthFooter } from './AuthFooter/AuthFooter';
export { AuthHeader } from './AuthHeader/AuthHeader';
export { AuthSocials } from './AuthSocials/AuthSocials';
export { EmailInput } from './EmailInput/EmailInput';
export { PasswordInput } from './PasswordInput/PasswordInput';
