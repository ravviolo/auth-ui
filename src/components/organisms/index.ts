export * from './LoginForm/LoginForm';
export * from './SignUpForm/SignUpForm';
