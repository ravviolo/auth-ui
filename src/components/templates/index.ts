export * from './LoginTemplate/LoginTemplate';
export * from './SignUpTemplate/SignUpTemplate';
