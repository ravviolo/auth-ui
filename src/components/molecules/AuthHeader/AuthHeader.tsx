import { Heading } from 'components/atoms';

interface Props {
  title: string;
}

export const AuthHeader = ({ title }: Props) => {
  return (
    <header>
      <Heading title={title} testId="auth-heading" />
    </header>
  );
};
