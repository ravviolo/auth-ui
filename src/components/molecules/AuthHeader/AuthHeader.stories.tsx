import { ComponentMeta, ComponentStory } from '@storybook/react';

import { AuthHeader } from 'components/molecules';

export default {
  title: 'Molecules/AuthHeader',
  component: AuthHeader,
  args: {
    title: 'Header Title',
  },
} as ComponentMeta<typeof AuthHeader>;

export const Default: ComponentStory<typeof AuthHeader> = (args) => (
  <AuthHeader {...args} />
);
Default.storyName = 'AuthHeader';
