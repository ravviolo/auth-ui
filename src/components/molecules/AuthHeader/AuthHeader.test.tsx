import { render, screen, within } from '@testing-library/react';
import { create } from 'react-test-renderer';

import { AuthHeader } from 'components/molecules';

import { Props as AuthHeaderProps } from './AuthHeader';

describe('AuthHeader', () => {
  const props: AuthHeaderProps = {
    title: 'Test Header Title',
    testId: 'test-auth-header',
  };

  it('renders header with heading text passed as a prop', () => {
    render(<AuthHeader {...props} />);

    const authHeader = screen.getByTestId('test-auth-header');

    const authHeading = within(authHeader).getByTestId('auth-heading');

    expect(authHeading).toHaveTextContent('Test Header Title');
  });
  it('matches snapshot', () => {
    const tree = create(<AuthHeader {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
