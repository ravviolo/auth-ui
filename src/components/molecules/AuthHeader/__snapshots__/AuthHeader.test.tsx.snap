// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthHeader matches snapshot 1`] = `
<header
  data-testid="test-auth-header"
>
  <h1
    className="sc-gKXOVf zAjwy"
    data-testid="auth-heading"
  >
    Test Header Title
  </h1>
</header>
`;
